File: Fitoplancton/AguasLastreConteoFito .vb
' Este codigo ha sido generado por el modulo psexport 20180802-w32 de PSeInt.
' Es posible que el codigo generado no sea completamente correcto. Si encuentra
' errores por favor reportelos en el foro (http://pseint.sourceforge.net).

Module CONTEOXCAMARASEDGWICKRAFTER

	Sub Main()
		Dim alturared As Double
		Dim cell As Double
		Dim celml As Double
		Dim diametrored As Double
		Dim n As Double
		Dim nml As Double
		Dim pi As Double
		Dim v1 As Double
		Dim v2 As Double
		Dim v3 As Double
		Dim volfiltradatotal As Double
		Dim volredlitros As Double
		Dim volumenred As Double
		' número total de Células contados por mililitro
		' número de Células contados en la cámara
		' Número de Células por litro
		' Número de Células por mililitro
		' Volumen concentrado de la muestra en mililitros
		' Volumen total de la alícuota en mililitros
		' Volumen de agua filtrada por la red en litros
		' Volumen de agua filtrada por la red en litros
		' Volumen de la red de muestro cm3
		' Volumen de la red de muestro en litros
		' Radio de la red en 
		' Largo total de la red
		Console.WriteLine("¿Cúal es la altura de la red en cm?")
		alturared = Double.Parse(Console.ReadLine())
		Console.WriteLine("¿Cúal es el diametro de la red en cm?")
		diametrored = Double.Parse(Console.ReadLine())
		volumenred = 1/3*Math.Pi*(diametrored/2)^2*alturared
		volredlitros = volumenred/1000
		Console.WriteLine("¿Cuál es el volumen de agua filtrada por la red en litros?")
		v3 = Double.Parse(Console.ReadLine())
		volfiltradatotal = v3+volredlitros
		Console.WriteLine("El volumen de la red es: ",volredlitros," litros")
		Console.WriteLine("El volumen de agua filtrada es: ",volfiltradatotal," litros")
		Console.WriteLine("¿Cúantas alícuoatas contó?")
		v2 = Double.Parse(Console.ReadLine())
		Console.WriteLine("¿Cuál es el número total de Células contados en todas las alícuotas?")
		n = Double.Parse(Console.ReadLine())
		' Número promedio total de Células contados por mililitro
		nml = n/v2
		Console.WriteLine("El número promedio total de Células de fitoplancton contados es: ",Math.Round(nml)," Cel/mL")
		Console.WriteLine("¿Cuál es el volumen concentrado de la muestra en mililitros?")
		v1 = Double.Parse(Console.ReadLine())
		' Densidad de Células por litro
		cell = (nml*v1)/(v2*volfiltradatotal)
		' Densidad de Células por mililitro
		celml = cell/1000
		Console.WriteLine("El resultado de la densidad de Células de fitoplancton es: ",Math.Round(cell)," Cel/L")
		Console.WriteLine("El resultado de la densidad de Células de fitoplancton es: ",Math.Round(celml)," Cel/ml")
	End Sub

End Module

File: Fitoplancton/AguasLastreConteoFito .py
# Este codigo ha sido generado por el modulo psexport 20180802-w32 de PSeInt.
# Es posible que el codigo generado no sea completamente correcto. Si encuentra
# errores por favor reportelos en el foro (http://pseint.sourceforge.net).

from math import pi

if __name__ == '__main__':
	# numero total de Cululas contados por mililitro
	nml = float()
	# numero de Cululas contados en la cumara
	n = float()
	# Numero de Cululas por litro
	cell = float()
	# Numero de Cululas por mililitro
	celml = float()
	# Volumen concentrado de la muestra en mililitros
	v1 = float()
	# Volumen total de la alucuota en mililitros
	v2 = float()
	# Volumen de agua filtrada por la red en litros
	v3 = float()
	# Volumen de agua filtrada por la red en litros
	volfiltradatotal = float()
	# Volumen de la red de muestro cm3
	volumenred = float()
	# Volumen de la red de muestro en litros
	volredlitros = float()
	# Radio de la red en 
	diametrored = float()
	# Largo total de la red
	
	print("¿Cúal es la altura de la red en cm?")
	alturared = float(input())
	print("¿Cúal es el diametro de la red en cm?")
	diametrored = float(input())
	volumenred = 1/3*pi*(diametrored/2)**2*alturared
	volredlitros = volumenred/1000
	print("¿Cúal es el volumen de agua filtrada por la red en litros?")
	v3 = float(input())
	volfiltradatotal = v3+volredlitros
	print("El volumen de la red es: ",volredlitros," litros")
	print("El volumen de agua filtrada es: ",volfiltradatotal," litros")
	print("¿Cuantas alícuotas contó?")
	v2 = float(input())
	print("¿Cúal es el numero total de Células contados en todas las alícuotas?")
	n = float(input())
	# Numero promedio total de Cululas contados por mililitro
	nml = n/v2
	print("El numero promedio total de Cululas de fitoplancton contados es: ",round(nml)," Cel/mL")
	print("¿Cúal es el volumen concentrado de la muestra en mililitros?")
	v1 = float(input())
	# Densidad de Cululas por litro
	cell = (nml*v1)/(v2*volfiltradatotal)
	# Densidad de Cululas por mililitro
	celml = cell/1000
	print("El resultado de la densidad de Cululas de fitoplancton es: ",round(cell)," Cel/L")
	print("El resultado de la densidad de Cululas de fitoplancton es: ",round(celml)," Cel/ml")

